src/interpreterstate.h

```diff
@@ -0,0 +1,207 @@
+/***********************************************************************
+ **
+ *W  interpreterstate.h      GAP source                 Markus Pfeiffer
+ **
+ **
+ ** This file declares all variables that are considered state for the
+ ** interpreter
+ **
+ */
+#ifndef GAP_INTERPRETER_STATE_H
+#define GAP_INTERPRETER_STATE_H
+
+#include <stdint.h>
+
+#include "system.h"
+#include "code.h"
+#include "scanner.h"
+#include "gasman.h"
+
+typedef struct InterpreterState
+{
+  /* From intrprtr.c */
+  Obj IntrResult;
+  UInt IntrIgnoring;
+  UInt IntrReturning;
+  UInt IntrCoding;
+  Obj IntrState;
+  Obj StackObj;
+  Int CountObj;
+  UInt PeriodicCheckCount;
+
+  /* From gvar.c */
+  Obj CurrNamespace;
+
+  /* From vars.c */
+  Bag BottomLVars;
+  Bag CurrLVars;
+  Obj *PtrLVars;
+  Bag LVarsPool[16];
+
+  /* From read.c */
+  syJmp_buf ReadJmpError;
+  syJmp_buf threadExit;
+  Obj StackNams;
+  UInt CountNams;
+  UInt ReadTop;
+  UInt ReadTilde;
+  UInt CurrLHSGVar;
+  UInt CurrentGlobalForLoopVariables[100];
+  UInt CurrentGlobalForLoopDepth;
+  Obj ExprGVars;
+  Obj ErrorLVars;
+  Obj ErrorLVars0;
+  Obj ReadEvalResult;
+
+  /* From scanner.c */
+  Char Value[MAX_VALUE_LEN];
+  UInt ValueLen;
+  UInt NrError;
+  UInt NrErrLine;
+  UInt            Symbol;
+  Char *          Prompt;
+  TypInputFile   InputFiles[16];
+  TypOutputFile OutputFiles[16];
+  int InputFilesSP;
+  int OutputFilesSP;
+  TypInputFile *  Input;
+  Char *          In;
+  TypOutputFile * Output;
+  TypOutputFile * InputLog;
+  TypOutputFile * OutputLog;
+  TypInputFile *  TestInput;
+  TypOutputFile * TestOutput;
+  TypOutputFile * IgnoreStdoutErrout;
+  Obj		  DefaultOutput;
+  Obj		  DefaultInput;
+  Char            TestLine [256];
+  TypOutputFile LogFile;
+  TypOutputFile LogStream;
+  TypOutputFile InputLogFile;
+  TypOutputFile InputLogStream;
+  TypOutputFile OutputLogFile;
+  TypOutputFile OutputLogStream;
+  Int HELPSubsOn;
+  Int NoSplitLine;
+  KOutputStream TheStream;
+  Char *TheBuffer;
+  UInt TheCount;
+  UInt TheLimit;
+
+  /* From exprs.c */
+  Obj (**CurrEvalExprFuncs)(Expr);
+
+  /* From stats.c */
+  Stat CurrStat;
+  Obj ReturnObjStat;
+  UInt (**CurrExecStatFuncs)(Stat);
+
+  /* From code.c */
+  Stat *PtrBody;
+  Stat OffsBody;
+  Stat *OffsBodyStack;
+  UInt OffsBodyCount;
+  UInt LoopNesting;
+  UInt *LoopStack;
+  UInt LoopStackCount;
+
+  Obj CodeResult;
+  Bag StackStat;
+  Int CountStat;
+  Bag StackExpr;
+  Int CountExpr;
+  Bag CodeLVars;
+
+  /* From funcs.h */
+  Int RecursionDepth;
+  Obj ExecState;
+
+  /* From opers.c */
+  Obj MethodCache;
+  Obj *MethodCacheItems;
+  UInt MethodCacheSize;
+  UInt CacheIndex;
+
+  /* From cyclotom.c */
+  Obj ResultCyc;
+  Obj  LastECyc;
+  UInt LastNCyc;
+
```

Comment:
I think I would prefer GlobalState, or some such.

Once we have got GAP<->HPCGAP merged, it would make sense to split this GlobalState into multiple pieces, but I think the initial merging should just be "one big blob", like HPC-GAP.